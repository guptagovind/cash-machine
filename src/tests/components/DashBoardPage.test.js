import React from 'react';
import {shallow} from 'enzyme';
import DashboardPage from '../../components/DashBoardPage';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<DashboardPage/>);
});

test('should render the dashboard', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should give notes of 20 and 10 for amount 30', () => {
  const value = 30.00;
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('amount')).toBe(value);
  wrapper.find('form').simulate('submit', {
    preventDefault() {
    }
  });
  expect(wrapper.state('results')).toEqual([20, 10]);
});

test('should give notes of 50, 20 & 10 for amount 80', () => {
  const value = 80.00;
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('amount')).toBe(value);
  wrapper.find('form').simulate('submit', {
    preventDefault() {
    }
  });
  expect(wrapper.state('results')).toEqual([50, 20, 10]);
});

test('should throw NoteUnavailableException', () => {
  const value = 125;
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('amount')).toBe(value);
  wrapper.find('form').simulate('submit', {
    preventDefault() {
    }
  });
  expect(wrapper.state('results')).toBe('throw NoteUnavailableException');
});

test('should throw InvalidArgumentException', () => {
  const value = -130.00;
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('amount')).toBe(value);
  wrapper.find('form').simulate('submit', {
    preventDefault() {
    }
  });
  expect(wrapper.state('results')).toBe('throw InvalidArgumentException');
});

test('should return result of Empty Set', () => {
  const value = "NULL";
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  });
  expect(wrapper.state('amount')).toBe(value);
  wrapper.find('form').simulate('submit', {
    preventDefault() {
    }
  });
  expect(wrapper.state('results')).toEqual(['Empty Set']);
});


