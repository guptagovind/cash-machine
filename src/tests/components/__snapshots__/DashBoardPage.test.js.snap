// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dashboard 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Cash Machine
    </h1>
    <p>
      Time to withdraw some cash.
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Enter amount"
        type="text"
        value=""
      />
      <div>
        <button
          className="button"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
    <div
      className="content-container"
    />
  </div>
</div>
`;
